### DISCUSSION TITLE
Does settings() work with GyroDriveBase?

### DISCUSSION THREAD

**Discussion by JuliaFiMSFT** [pybricks/support#1097] [General]
I am using v3.3.0b6 (Pybricks Beta v2.2.0-beta.5) and the GyroDriveBase works as expected. However, I cannot use drive_base.settings(). In this code, I get Invalid Argument.

```python
from pybricks.hubs import PrimeHub
from pybricks.pupdevices import Motor, ColorSensor, UltrasonicSensor, ForceSensor
from pybricks.parameters import Button, Color, Direction, Port, Side, Stop
from pybricks.robotics import GyroDriveBase
from pybricks.tools import wait, StopWatch

hub = PrimeHub()

left_wheel = Motor(Port.B, Direction.COUNTERCLOCKWISE)
right_wheel = Motor(Port.F)

drive_base = GyroDriveBase(left_wheel, right_wheel, wheel_diameter=55.8, axle_track=111)
drive_base.settings(1000, 1000, 1000, 1000) #this produces and Invalid argument error
drive_base.distance_control.target_tolerances(speed=None, position=None)
drive_base.heading_control.target_tolerances(speed=None, position=None)

print(hub.ble.version())
drive_base.straight(500, then=Stop.HOLD)
```

**Comment by laurensvalk** [comment]
The `settings()` method is available, but this error happens when the values are higher than the motors could possibly deliver.

We added this check internally to give immediate feedback if something isn't right (as opposed to giving unexpected runtime behavior).

But we could certainly make the message much clearer.

We have this for port errors, for example:

```
OSError: [Errno 19] ENODEV: 

A sensor or motor is not connected to the specified port:
--> Check the cables to each motor and sensor.
--> Check the port settings in your script.
--> Check the line in your script that matches
    the line number given in the 'Traceback' above.
```

For invalid arguments, perhaps we should have something like:

```
ValueError: Invalid argument

One of the provided values is too high or too low.
--> Check the line in your script that matches
    the line number given in the 'Traceback' above.
--> Check if the given values are not too high or too low.
```

For this method specifically, we could perhaps print what the maximum values really are, since it depends on the wheel diameter and axle track. (A wide robot with small wheels can't turn quite as fast, for example.)

**Reply by JuliaFiMSFT** [reply]
Ahhh, I see, that fixes it! If you could print the maximum values that would be very helpful. For the robot that my younger team built, the maximum straight_speed is 900. They of course set everything to 1000 and then got confused. Their non-gyro version of their code "worked" for them then switching to the gyro version was confusing.
